libs/SalesforceSDKCore/SalesforceSDKCore/Classes/URLHandlers/SFSDKIDPRequestHandler.m

```diff
@@ -0,0 +1,53 @@
+/*
+ SFSDKIDPRequestHandler.m
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 8/28/17.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import "SFSDKURLHandler.h"
+#import "SFSDKIDPRequestHandler.h"
+#import "SFOAuthCredentials.h"
+#import "SFSDKOAuthClient.h"
+#import "SFSDKOAuthClientConfig.h"
+#import "SFSDKUserSelectionNavViewController.h"
+#import "SFSDKIDPAuthClient.h"
+#import "SFUserAccountManager+URLHandlers.h"
+
+@implementation SFSDKIDPRequestHandler
+
+- (BOOL)canHandleRequest:(NSURL *)url options:(NSDictionary *)options {
+    NSRange clientIdRange = [url.absoluteString rangeOfString:@"oauth_client_id="];
+    NSRange codeVerifierRange = [url.absoluteString rangeOfString:@"code_verifier="];
+    NSRange redirectRange = [url.absoluteString rangeOfString:@"oauth_redirect_uri="];
+    return (codeVerifierRange.location!=NSNotFound) &&
+    (clientIdRange.location!=NSNotFound)  &&
+    (redirectRange.location!=NSNotFound);
+}
+
+- (BOOL)processRequest:(NSURL *)url options:(NSDictionary *)options {
+    [[SFUserAccountManager sharedInstance] handleIdpRequest:url options:options];
```

Comment:
Should SFSDKIDPRequestHandler handle parsing and pass the extracted values to handleIdpRequest?